opencog/cython/PythonEval.cc

```diff
@@ -44,60 +52,239 @@ using namespace opencog;
+
+    logger().debug("[global_python_initialize] Adding OpenCog sys.path "
+            "directories");
+
+    // Get starting "sys.path".
+    PyRun_SimpleString(
+                "import sys\n"
+                "import StringIO\n"
+                );
+    PyObject* pySysPath = PySys_GetObject((char*)"path");
+
+    // Add default OpenCog module directories to the Python interprator's path.
+    const char** config_paths = DEFAULT_PYTHON_MODULE_PATHS;
+    for (int i = 0; config_paths[i] != NULL; ++i) {
+        boost::filesystem::path modulePath(config_paths[i]);
+        if (boost::filesystem::exists(modulePath)) {
+            const char* modulePathCString = modulePath.string().c_str();
+            PyObject* pyModulePath = PyBytes_FromString(modulePathCString);
+            PyList_Append(pySysPath, pyModulePath);
+            Py_DECREF(pyModulePath);
+        }
     }
+
+    // Add custom paths for python modules from the config file.
+    if (config().has("PYTHON_EXTENSION_DIRS")) {
+        std::vector<std::string> pythonpaths;
+        // For debugging current path
+        tokenize(config()["PYTHON_EXTENSION_DIRS"], 
+                std::back_inserter(pythonpaths), ", ");
+        for (std::vector<std::string>::const_iterator it = pythonpaths.begin();
+             it != pythonpaths.end(); ++it) {
+            boost::filesystem::path modulePath(*it);
+            if (boost::filesystem::exists(modulePath)) {
+                const char* modulePathCString = modulePath.string().c_str();
+                PyObject* pyModulePath = PyBytes_FromString(modulePathCString);
+                PyList_Append(pySysPath, pyModulePath);
+                Py_DECREF(pyModulePath);
+            } else {
+                logger().warn("PythonEval::%s Could not find custom python"
+                        " extension directory: %s ",
+                        __FUNCTION__, (*it).c_str() );
+            }
+        }
+    }
+
+    // NOTE: Can't use get_path_as_string() yet because it is defined in a
+    // Cython api which we can't import unless the sys.path is correct. So
+    // we'll write it out before the imports below to aid in debugging.
+    if (logger().isDebugEnabled()) {
+        logger().debug("Python 'sys.path' after OpenCog config adds is:");
+        Py_ssize_t pathSize = PyList_Size(pySysPath);
+        for (int pathIndex = 0; pathIndex < pathSize; pathIndex++) {
+            PyObject* pySysPathLine = PyList_GetItem(pySysPath, pathIndex);
+            const char* sysPathCString = PyString_AsString(pySysPathLine);
+            logger().debug("    %2d > %s", pathIndex, sysPathCString);
+            // NOTE: PyList_GetItem returns borrowed reference so don't do this:
+            // Py_DECREF(pySysPathLine);
+        } 
+    }
+
+    // Initialize the auto-generated Cython api. Do this AFTER the python
+    // sys.path is updated so the imports can find the cython modules.
+    import_opencog__atomspace();
+    import_opencog__agent_finder();
```

Comment:
You may need agent-finder for now, but in the long run, I'd like to see a version that doesn't drag agents into the mix, because the long-term strategy is to not use agents -- they're "deprecated"